src/System.Windows.Forms/src/System/Windows/Forms/RadioButton.RadioButtonAccessibleObject.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class RadioButton
+    {
+        public class RadioButtonAccessibleObject : ControlAccessibleObject
+        {
+            private readonly RadioButton _owningRadioButton;
+
+            public RadioButtonAccessibleObject(RadioButton owner) : base(owner)
+            {
+                _owningRadioButton = owner;
+            }
+
+            public override string DefaultAction
+            {
+                get
+                {
+                    string defaultAction = Owner.AccessibleDefaultActionDescription;
+                    if (defaultAction != null)
+                    {
+                        return defaultAction;
+                    }
+
+                    return SR.AccessibleActionCheck;
+                }
+            }
+
+            public override AccessibleRole Role
+            {
+                get
+                {
+                    AccessibleRole role = Owner.AccessibleRole;
+                    if (role != AccessibleRole.Default)
+                    {
+                        return role;
+                    }
+
+                    return AccessibleRole.RadioButton;
+                }
+            }
+
+            public override AccessibleStates State
+            {
+                get
+                {
+                    if (_owningRadioButton.Checked)
+                    {
+                        return AccessibleStates.Checked | base.State;
+                    }
+
+                    return base.State;
+                }
+            }
+
+            internal override bool IsItemSelected
+                => _owningRadioButton.Checked;
+
+            internal override UiaCore.ToggleState ToggleState
+                => _owningRadioButton.Checked ? UiaCore.ToggleState.On : UiaCore.ToggleState.Off;
+
+            public override void DoDefaultAction()
+            {
+                if (_owningRadioButton.IsHandleCreated)
+                {
+                    _owningRadioButton.PerformClick();
+                }
+            }
+
+            internal override object GetPropertyValue(UiaCore.UIA propertyID)
+                => propertyID switch
+                {
+                    UiaCore.UIA.NamePropertyId
+                        => Name,
+                    UiaCore.UIA.AutomationIdPropertyId
+                        => _owningRadioButton.IsHandleCreated ? _owningRadioButton.Name : string.Empty,
```

Comment:
Access to the `Name` doesn't require a handle, nor it can be `null`
```suggestion
                        => Owner.Name,
```

Reply:
Thanks, fixed.